[Flaking Test] TestUpdateAllocatedResourcesStatus

### Which jobs are flaking?

https://testgrid.k8s.io/presubmits-kubernetes-blocking#pull-kubernetes-unit

### Which tests are flaking?

TestUpdateAllocatedResourcesStatus

### Since when has it been flaking?

Few days ago

### Testgrid link

https://testgrid.k8s.io/presubmits-kubernetes-blocking#pull-kubernetes-unit

### Reason for failure (if possible)

_No response_

### Anything else we need to know?

_No response_

### Relevant SIG(s)

/sig node

cc @ffromani 

```
I0725 16:04:21.570321  958866 manager.go:510] "Failed to read data from checkpoint" checkpoint="kubelet_internal_checkpoint" err="checkpoint is not found"
I0725 16:04:21.606742  958866 fake_topology_manager.go:33] "NewFakeManager"
--- FAIL: TestUpdateAllocatedResourcesStatus (0.00s)
    manager_test.go:1931: UpdateAllocatedResourcesStatus failed, expected: [{test-container {nil nil nil} {nil nil nil} false 0    <nil> map[] nil [] nil [{test-resource [{dev1 Healthy} {dev2 Unhealthy}]}]}], got: [{test-container {nil nil nil} {nil nil nil} false 0    <nil> map[] nil [] nil [{test-resource [{dev2 Unhealthy} {dev1 Healthy}]}]}]
```

I am unfamilar with this code so wanted to see if you have any ideas on this.

> cc @ffromani
> 
> ```
> I0725 16:04:21.570321  958866 manager.go:510] "Failed to read data from checkpoint" checkpoint="kubelet_internal_checkpoint" err="checkpoint is not found"
> I0725 16:04:21.606742  958866 fake_topology_manager.go:33] "NewFakeManager"
> --- FAIL: TestUpdateAllocatedResourcesStatus (0.00s)
>     manager_test.go:1931: UpdateAllocatedResourcesStatus failed, expected: [{test-container {nil nil nil} {nil nil nil} false 0    <nil> map[] nil [] nil [{test-resource [{dev1 Healthy} {dev2 Unhealthy}]}]}], got: [{test-container {nil nil nil} {nil nil nil} false 0    <nil> map[] nil [] nil [{test-resource [{dev2 Unhealthy} {dev1 Healthy}]}]}]
> ```
> 
> I am unfamilar with this code so wanted to see if you have any ideas on this.

yes, I'll have a look ASAP

--- a/pkg/kubelet/cm/devicemanager/manager_test.go
+++ b/pkg/kubelet/cm/devicemanager/manager_test.go
@@ -22,6 +22,7 @@ import (
 	"path/filepath"
 	"reflect"
 	goruntime "runtime"
+	"sort"
 	"sync"
 	"sync/atomic"
 	"testing"
@@ -1927,7 +1928,28 @@ func TestUpdateAllocatedResourcesStatus(t *testing.T) {
 			AllocatedResourcesStatus: []v1.ResourceStatus{expectedStatus},
 		},
 	}
+
+	// Sort the resources for the expected status and actual status
+	sortContainerStatuses(status.ContainerStatuses)
+	sortContainerStatuses(expectedContainerStatuses)
+
 	if !reflect.DeepEqual(status.ContainerStatuses, expectedContainerStatuses) {
 		t.Errorf("UpdateAllocatedResourcesStatus failed, expected: %v, got: %v", expectedContainerStatuses, status.ContainerStatuses)
+	}
+}
+
+// Helper function to sort ResourceHealth slices
+func sortResourceHealth(resources []v1.ResourceHealth) {
+	sort.SliceStable(resources, func(i, j int) bool {
+		return resources[i].ResourceID < resources[j].ResourceID
+	})
+}
+
+// Helper function to sort ContainerStatus slices
+func sortContainerStatuses(statuses []v1.ContainerStatus) {
+	for i := range statuses {
+		for j := range statuses[i].AllocatedResourcesStatus {
+			sortResourceHealth(statuses[i].AllocatedResourcesStatus[j].Resources)
+		}
 	}
 }

